internal/basecomponent/base.go

```diff
@@ -157,6 +154,20 @@ func (b *BaseDendrite) CreateHTTPFederationSenderAPIs() federationSenderAPI.Fede
 	return f
 }
 
+// CreateHTTPFederationSenderAPIs returns FederationSenderInternalAPI for hitting
+// the federation sender over HTTP
+func (b *BaseDendrite) CreateHTTPServerKeyAPIs() serverKeyAPI.ServerKeyInternalAPI {
+	f, err := serverKeyAPI.NewServerKeyInternalAPIHTTP(
+		b.Cfg.FederationSenderURL(),
```

Comment:
This is probably wrong?

Reply:
Ah, yes it is. That was masked by the fact that we overwrite those all to be the same endpoint in monolith mode. Good catch.